### PR TITLE
Restrict flare file from being accessible by other users on Unix

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Restrict flare file from being accessible by other users on Unix

### Describe how to test/QA your changes

Create and send from from both Linux and Windows when the agent is running and not running.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/util/filesystem/permission_nowindows.go [hunk 1]
```diff
@@ -54,3 +56,18 @@ func (p *Permission) RestrictAccessToUser(path string) error {
 
 	return nil
 }
+
+// RemoveAccessToOtherUsers on Unix this calls RestrictAccessToUser and then removes all access to the file for 'group'
+// and 'other'
+func (p *Permission) RemoveAccessToOtherUsers(path string) error {
```
Comment (Kaderinho) [inline pkg/util/filesystem/permission_nowindows.go:62]: It's mostly related to the function above but why are we using a `Permission` here (I mean on `RestrictAccessToUser`). Is it a legacy code that has been changed and the argument `p *Permission` has never been removed ?

File: pkg/util/filesystem/permission_nowindows.go [hunk 2]
```diff
@@ -54,3 +56,18 @@ func (p *Permission) RestrictAccessToUser(path string) error {
...
+	// We first try to set other and group to "dd-agent" when possible
+	_ = p.RestrictAccessToUser(path)
+
+	fperm, err := os.Stat(path)
+	if err != nil {
+		return err
+	}
+	// We keep the original 'user' rights but set 'group' and 'use' to zero.
```
Comment (Kaderinho) [inline pkg/util/filesystem/permission_nowindows.go:70]:
```suggestion
	// We keep the original 'user' rights but set 'group' and 'other' to zero.
```

**Review by Kaderinho — APPROVED** [review]
LGTM, feel free to approve after addressing the `newPerm` formula

File: pkg/util/filesystem/permission_nowindows.go (hunk 1 above)
Comment (hush-hush) [reply pkg/util/filesystem/permission_nowindows.go:62]: It's for Windows. The Windows implementation need to pull a few IDs and ACL ( [here](https://github.com/DataDog/datadog-agent/blob/main/pkg/util/filesystem/permission_windows.go#L27) ).

Comment (Kaderinho) [reply pkg/util/filesystem/permission_nowindows.go:62]: Ohhh, thanks 🙇

**Pull request merged**